numpy/lib/ufunclike.py

```diff
@@ -138,6 +139,10 @@ def isposinf(x, out=None):
     array([0, 0, 1])
 
     """
+    x = nx.asanyarray(x)
```

Comment:
As @eric-wieser notes, this leads to a change of behaviour for objects that implement `__array_ufunc__` but cannot be converted to arrays.